htlcswitch/circuit.go

```diff
@@ -53,82 +80,220 @@ func newPaymentCircuit(src, dest lnwire.ShortChannelID, key circuitKey,
 		Src:         src,
 		Dest:        dest,
 		PaymentHash: key,
-		RefCount:    1,
 		Obfuscator:  obfuscator,
 	}
 }
 
 // isEqual checks the equality of two payment circuits.
-func (a *paymentCircuit) isEqual(b *paymentCircuit) bool {
-	return bytes.Equal(a.PaymentHash[:], b.PaymentHash[:]) &&
-		a.Src == b.Src &&
-		a.Dest == b.Dest
+func (c *paymentCircuit) isEqual(b *paymentCircuit) bool {
+	return bytes.Equal(c.PaymentHash[:], b.PaymentHash[:]) &&
+		c.Src == b.Src &&
+		c.Dest == b.Dest
+}
+
+// Decode reads the data from the byte stream and initialize the
+// payment circuit with data.
+func (c *paymentCircuit) Decode(r io.Reader) error {
+	if _, err := r.Read(c.PaymentHash[:]); err != nil {
+		return err
+	}
+
+	var srcBytes [8]byte
+	if _, err := r.Read(srcBytes[:]); err != nil {
```

Comment:
can we always assume `r` only contains 8 bytes? `r.Read()` will also return the number of types read (as first argument). Shouldn't we assert that it's 8?